ui/shared/PushHealthStatus.jsx

```diff
@@ -4,39 +4,41 @@ import { Badge } from 'reactstrap';
 
 import PushModel from '../models/push';
 import { getPushHealthUrl } from '../helpers/url';
+import { didObjectsChange } from '../helpers/object';
 
 class PushHealthStatus extends Component {
   constructor(props) {
     super(props);
 
     this.state = {
-      healthStatus: 'Loading...',
+      healthStatus: '',
       needInvestigation: 0,
     };
   }
 
   async componentDidMount() {
-    await this.loadLatestStatus();
+    const {
+      jobCounts: { completed },
+    } = this.props;
+
+    if (completed > 0) {
+      await this.loadLatestStatus();
+    }
   }
 
-  shouldComponentUpdate(nextProps, nextState) {
+  async componentDidUpdate(prevProps) {
     const { jobCounts } = this.props;
-    const { healthStatus } = this.state;
-
-    return (
-      jobCounts !== nextProps.jobCounts ||
-      healthStatus !== nextState.healthStatus
-    );
-  }
+    const fields = ['completed', 'fixedByCommit', 'pending', 'running'];
 
-  async componentDidUpdate() {
-    await this.loadLatestStatus();
+    if (didObjectsChange(jobCounts, prevProps.jobCounts, fields)) {
+      await this.loadLatestStatus();
+    }
   }
 
   async loadLatestStatus() {
-    const { repoName, pushId } = this.props;
+    const { repoName, revision } = this.props;
 
-    PushModel.getHealthSummary(repoName, pushId).then(resp => {
+    PushModel.getHealthSummary(repoName, revision).then(resp => {
```

Comment:
I'm curious why this is using `then` instead of the `getData` wrapper?

Reply:
dang, just a mistake.  Fixed it...